components/dashboard/src/hooks/use-check-dedicated-setup.ts

```diff
@@ -0,0 +1,23 @@
+/**
```

Comment:
For custom hooks which are quite ubiquitous, I would personally prefer to group by domain topic rather than technically. That means I suggest moving this hook to `DedicatedSetup`